src/tactic/core.lean

```diff
@@ -1496,41 +1481,73 @@ The number 2 indicates that first the two bound variables of the
 rather than `=ₐ` or `is_def_eq` tells us that proof script should
 not see the difference between the two.
  -/
-meta def goal := ℕ × expr
+meta def user_visible_goal := ℕ × expr
 
 /-- proof state made of multiple `goal` meant for comparing
 the result of running different tactics -/
-meta def proof_state := list goal
+meta def proof_state := list user_visible_goal
 
-meta instance goal.inhabited : inhabited goal := ⟨(0,var 0)⟩
+meta instance goal.inhabited : inhabited user_visible_goal := ⟨(0,var 0)⟩
 meta instance proof_state.inhabited : inhabited proof_state :=
-(infer_instance : inhabited (list goal))
+(infer_instance : inhabited (list user_visible_goal))
 
-/--
-Run `tac` in a disposable proof state and return the state.
-See `proof_state` and `goal`.
+meta def get_user_visible_goal : tactic user_visible_goal :=
+retrieve $ do
+ls ← local_context,
+     tgt ← target >>= instantiate_mvars >>= pis ls,
+     pure (ls.length, tgt)
+
+meta def goal_of_mvar (g : expr) : tactic user_visible_goal :=
+with_local_goals' [g] get_user_visible_goal
+
+/-- `get_proof_state` lists the user visible goal for each goal
+    of the current state and for each goal, abstracts all of the
```

Comment:
I'm concerned about the indentation here (and in other docstrings). I haven't taken a look at this file specifically in VS Code, but 4 spaces in markdown signifies a code block so the rendering could get screwed up.

Reply:
This is what I get when I open it in VS Code:

<img width="533" alt="Screen Shot 2020-03-25 at 2 16 35 AM" src="https://user-images.githubusercontent.com/71008/77507473-e6f40180-6e3e-11ea-9671-8cd3d33a7044.png">

Is this what you'd expect?

Reply:
Sorry, what I meant is how the docstring renders in a hover or e.g. the mathlib docs. This is what I see with the current indentation:
<img width="499" alt="Screenshot 2020-03-25 03 22 20" src="https://user-images.githubusercontent.com/5209952/77511665-f3c92300-6e47-11ea-9c0c-842e200f6cfb.png">

And it looks like this if I remove all the indentation:
<img width="344" alt="Screenshot 2020-03-25 03 24 33" src="https://user-images.githubusercontent.com/5209952/77511784-2d9a2980-6e48-11ea-9142-bdc7afdc625b.png">

Reply:
I'll try to make a suggestion.

Reply:
I applied your suggestion. Let me know if I should fix the formatting elsewhere.

Reply:
Thanks! The others look fine to me.